packages/commons-test/src/apiMocks.ts

```diff
@@ -0,0 +1,313 @@
+import {
+  attributeRegistryApi,
+  catalogApi,
+  delegationApi,
+  agreementApi,
+  purposeApi,
+  tenantApi,
+  authorizationApi,
+  m2mGatewayApi,
+  eserviceTemplateApi,
+} from "pagopa-interop-api-clients";
+import { generateMock } from "@anatine/zod-mock";
+import { generateId } from "pagopa-interop-models";
+import { z } from "zod";
+
+export function getMockedApiPurposeVersion({
+  state,
+}: {
+  state?: purposeApi.PurposeVersionState;
+} = {}): purposeApi.PurposeVersion {
+  return {
+    id: generateId(),
+    createdAt: new Date().toISOString(),
+    dailyCalls: generateMock(z.number().positive()),
+    state: state ?? purposeApi.PurposeVersionState.Enum.DRAFT,
+  };
+}
+
+export function getMockedApiPurpose({
+  versions,
+}: {
+  versions?: purposeApi.PurposeVersion[];
+} = {}): purposeApi.Purpose {
+  return {
+    id: generateId(),
+    eserviceId: generateId(),
+    consumerId: generateId(),
```

Comment:
To be more precise on typing.
```suggestion
    id: generateId<PurposeId>(),
    eserviceId: generateId<EServiceId>(),
    consumerId: generateId<TenantId>(),
```